proc_scripts/side_by_side.py

```diff
@@ -0,0 +1,48 @@
+from pyspecdata import *
+from sympy import symbols
+def expand_limits(thisrange,s):
+    thisrange = list(thisrange)
+    full_range = s.getaxis('t2')[r_[0,-1]]
+    retval = array([thisrange[j] if thisrange[j] is not
+            None else full_range[j] for j in [0,-1]])
+    print(repr(retval))
+    m = mean(retval)
+    s = m-retval
+    retval = 3*s+m
+    sgn = [-1,1] # greater than or less than
+    return tuple(full_range[j] if
+            retval[j]*sgn[j] > full_range[j]*sgn[j]
+            else
+            retval[j] for j in range(2))
+def draw_limits(thisrange,s):
+    full_range = s.getaxis('t2')[r_[0,-1]]
+    dw = diff(s.getaxis('t2')[:2]).item()
+    print("I find the full range to be",full_range)
+    sgn = [-1,1] # add or subtract
+    pairs = [[thisrange[j],full_range[j]+0.5*dw*sgn[j]] for j in range(2)]
+    pairs[0] = pairs[0][::-1] # flip first
+    my_xlim = gca().get_xlim() # following messes w/ xlim for some reason
+    for j in pairs:
+        if None not in j:
+            print("drawing a vspan at",j)
+            axvspan(j[0],j[1],color='w',alpha=0.5,linewidth=0)
+    gca().set_xlim(my_xlim)
+class fl_mod (figlist_var):
+    def side_by_side(self,plotname,s,thisrange):
+        """a bit of a hack to get the two subplots into
+        the figure list -- also a good test for objective
+        figure list -- for each slice out 3x thisrange, and then
+        show the lines for thisrange"""
+        thisfig,(ax1,ax2) = subplots(1,2)
+        fl.next(plotname, fig=thisfig)
+        sca(ax1)
+        forplot = s['t2':expand_limits(thisrange,s)]
+        fl.image(forplot)
+        print("drawing limits",thisrange)
+        draw_limits(thisrange,forplot)
+        sca(ax2)
+        fl.image(forplot.C.cropped_log())
+        draw_limits(thisrange,forplot)
+        title('cropped log')
+        return
+fl = fl_mod()    
```

Comment:
don't do this here -- this is creating an instance of the class

Reply:
done

Reply:
you don't want to create an instance inside the function either -- again, this goes back to reading in langtangen -- mark this as resolved when read

Reply:
So I will/am working through Langtangen as well as reading up on classes in python. I made a fix and I THINK this is what you want? If not I will try again after further reading

Reply:
no, you are currently creating an instance of fl_mod with `        fl=figlist_var()` inside the side by side function -- this is what I refer to with the comment from the 15th, above

Reply:
can we do this as just a function of side_by_side? Is there a reason we are making an entire class?

Reply:
yes: (1) that would be inconsistent with the way we are doing it (2) the way you're thinking about it is wrong.  You are not "making an entire class".  Rather, you are simply enhancing the fl_var class by adding new functions to it.

Since this is taking too long, I created a pull request for you to look at.

Please be very careful, though, not having done it yourself, I'm worried that you'll miss what's going on here.

Reply:
right so why not just add the function to figlist_var class?

Reply:
because that would involve modifying pyspecdata.  OK to resolve comment.